python/tests/test_balance_metrics.py

```diff
@@ -63,6 +74,16 @@ def b1_index_definition(tree):
     )
 
 
+def b2_index_definition(tree, base):
+    if tree.num_roots != 1:
+        raise ValueError("B2 index not defined for multiroot trees")
+    proba = [
+        np.prod([1 / tree.num_children(u) for u in nodes_to_root(tree, leaf)])
+        for leaf in tree.leaves()
+    ]
+    return -sum(p * math.log(p, base) for p in proba)
```

Comment:
Might as well stick with numpy ops here,
```
return -np.sum(proba * np.log(proba, base))
```

Reply:
Apparently, for numpy, `log` is always the natural logarithm (no base parameter, see [here](https://numpy.org/doc/stable/reference/generated/numpy.log.html)).